apis/apiextensions/v1/composition_patches.go

```diff
@@ -60,7 +69,17 @@ type PatchPolicy struct {
 	// +kubebuilder:validation:Enum=Optional;Required
 	// +optional
 	FromFieldPath *FromFieldPathPolicy `json:"fromFieldPath,omitempty"`
-	MergeOptions  *xpv1.MergeOptions   `json:"mergeOptions,omitempty"`
```

Comment:
We can't delete `MergeOptions` here in core Crossplane. It would be a breaking API change.

We could introduce `ToFieldPath`, support both, and add a comment to `MergeOptions` saying that it's deprecated.

Reply:
Regarding last line, just wondering how is it possible to replace MergeOptions with ToFieldPath without touching core? Since MergeOptions is a field inside PatchPolicy https://github.com/crossplane/crossplane/blob/b565aaccf9a9a2e87a20c693d476b8be81bd6994/apis/apiextensions/v1/composition_patches.go#L114

Reply:
See https://github.com/crossplane/crossplane/issues/5421#issuecomment-1985273029.

If you're trying to fix #5421, the point of that issue is to update tooling that people can use to migrate from native patch and transform to https://github.com/crossplane-contrib/function-patch-and-transform. The former uses `MergeOptions`, the latter uses `ToFieldPath` instead.

What you're doing here is instead changing the native P&T implementation to use `ToFieldPath` instead of `MergeOptions`. 

(P.S. Please follow the instructions in the PR template. Especially the part about using Fixes to make it clear what issue you're trying to fix.)

Reply:
@negz Thanks for the hint! I have limited my changes to converter.go. Its still in draft as I hnvt updated the tests.

Reply:
@negz I have now updated the converter file to migrate to ToFieldPath. I had to duplicate some of the structs to achieve this - let me know wdyt? I will fix the e2e tests and add couple of unit tests to cover different migration scenarios :)